### PR TITLE
Fixes l10n for brave_strings.grd.

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/36257

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/commands/lib/l10nUtil.js [hunk 1]
```diff
@@ -258,16 +258,22 @@ const l10nUtil = {
...
+      async ([sourcePath, destPath]) => {
+        console.log("Resetting " + path.relative(srcDir, destPath) + " <- " +
+          path.relative(srcDir, sourcePath))
+        let contents = await new Promise(
+          resolve => fs.readFile(sourcePath, 'utf8',
+            (err, data) => resolve(data)))
+        await new Promise(
+          resolve => fs.writeFile(destPath, contents, 'utf8', resolve))
+      }
+    )
```
Comment (mkarolin) [inline build/commands/lib/l10nUtil.js:276]: No functional changes here, just updated the comment and formatting.

File: script/chromium-rebase-l10n.py [hunk 2]
```diff
@@ -167,6 +174,15 @@ def generate_overrides_and_replace_strings(source_string_path):
...
     if len(modified_messages) > 0 or len(modified_parts) > 0:
+        # Fix output filenames to generate "brave" files instead of "chromium".
+        if os.path.basename(source_string_path) == 'brave_strings.grd':
+            for xtb_filename in modified_xml_tree.xpath(
+                "//file[re:test(@path, '.*\\.xtb')]",
+                namespaces={"re": "http://exslt.org/regular-expressions"}):
+                xtb_filename.attrib['path'] = \
+                    xtb_filename.attrib['path'].replace('chromium_strings',
+                                                        'brave_strings')
+        print(f'Writing override {override_string_path}')
```
Comment (mkarolin) [inline script/chromium-rebase-l10n.py:185]: Renamed for consistency.

File: script/lib/l10n/grd_utils.py [hunk 3]
```diff
@@ -151,9 +151,12 @@ def update_xtbs_locally(grd_file_path, brave_source_root):
     xtb_files = get_xtb_files(grd_file_path)
     chromium_grd_file_path = get_chromium_grd_src_with_fallback(grd_file_path,
         brave_source_root)
-    chromium_xtb_files = get_xtb_files(grd_file_path)
+    chromium_xtb_files = get_xtb_files(chromium_grd_file_path)
```
Comment (mkarolin) [inline script/lib/l10n/grd_utils.py:154]: This is the actual fix.

File: build/commands/lib/l10nUtil.js [hunk 4]
```diff
@@ -258,16 +258,22 @@ const l10nUtil = {
...
-    })
+    const ops = Object.entries(getChromiumToAutoGeneratedBraveMapping()).map(
+      async ([sourcePath, destPath]) => {
+        console.log("Resetting " + path.relative(srcDir, destPath) + " <- " +
+          path.relative(srcDir, sourcePath))
+        let contents = await new Promise(
+          resolve => fs.readFile(sourcePath, 'utf8',
+            (err, data) => resolve(data)))
+        await new Promise(
+          resolve => fs.writeFile(destPath, contents, 'utf8', resolve))
```
Comment (goodov) [inline build/commands/lib/l10nUtil.js:274]: do we really need to read/write manually?

The fs.copyFile() method is used to asynchronously copy a file from the source path to the destination path. By default, Node.js will overwrite the file if it already exists at the given destination.

**Review by goodov — APPROVED** [review]

File: build/commands/lib/l10nUtil.js (hunk 4 above)
Comment (mkarolin) [reply build/commands/lib/l10nUtil.js]: Nope, I don't think we do 😄

**Review by bsclifton — APPROVED** [review]
string reviewers ++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.65.35 Chromium: 122.0.6261.69 (Offizieller Build) nightly (64-Bit)
--- | ---
Überarbeitung | 8873b7ee8825b7b68a55a5ffbd3ab44759476d02
Betriebssystem | Windows 10 Version 22H2 (Build 19045.3803)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/36257#issue-2146609391, ensured that the strings mentioned via `brave://settings/security` & `brave://settings/content/v8` are correctly translated as per the following:

**Using `German` as the local for the verification**

**`Example`** | **`Example`**
----------------|----------------
germany1 | germany2

**Pull request merged**